lib/babe/verify_test.go

```diff
@@ -23,9 +23,234 @@ import (
 
 	"github.com/stretchr/testify/require"
 
+	"github.com/ChainSafe/gossamer/dot/state"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
 	"github.com/ChainSafe/gossamer/lib/crypto/sr25519"
+	"github.com/ChainSafe/gossamer/lib/genesis"
+	"github.com/ChainSafe/gossamer/lib/trie"
 )
 
+func newTestVerificationManager(t *testing.T, withBlock bool, descriptor *NextEpochDescriptor) *VerificationManager {
+	dbSrv := state.NewService("")
+	dbSrv.UseMemDB()
+
+	genesisData := new(genesis.Data)
+
+	err := dbSrv.Initialize(genesisData, genesisHeader, trie.NewEmptyTrie())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = dbSrv.Start()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if descriptor == nil {
+		descriptor = &NextEpochDescriptor{}
+	}
+
+	// currentEpoch = 2
```

Comment:
I think I had a comment in here before about explaining how this relates to the test pre-digest item. Might be nice to add it back and even pull it out as a test var or constant.